ykrt/src/compile/jitc_yk/mod.rs

```diff
@@ -72,8 +89,10 @@ impl Compiler for JITCYk {
             ));
         }
 
+        let sti = sti.map(|s| s.as_any().downcast::<YkSideTraceInfo>().unwrap());
```

Comment:
Another oblivious question: why is this `Any` and `downcast` dance needed?

Reply:
`YkSideTraceInfo` has data that the trait `SideTraceInfo` doesn't know about. We downcast so we can access that data. In the LLVM backend we made the mistake of trying to predict what a backend needs, and those predictions turned out to be wrong. So now `SideTraceInfo` is basically an opaque pointer that the backend can pass around without dictating what other backends might want to do.

Reply:
OK. Hopefully this hack can go if we delete the llvm backend.